client/handle_http.go

```diff
@@ -386,7 +396,26 @@ func checksumFromHttpDigest(httpDigest string) ChecksumType {
 	return AlgUnknown
 }
 
-func httpDigestFromChecksum(checksumType ChecksumType) string {
+// List all the checksum types known to the client
+func KnownChecksumTypes() (result []ChecksumType) {
+	result = make([]ChecksumType, AlgUnknown-algFirst)
+	for idx := algFirst; idx < AlgUnknown; idx++ {
+		result[idx-algFirst] = idx
+	}
+	return
+}
```

Comment:
This feels like a complex way to declare what could be a static list, and while I think your goal here was to prevent creating a list that needs to be maintained, this approach comes with its own assumptions. If you really think we need the iota math here, can you:
1. Leave a comment near the iota declaration that indicates the iota must not skip any integers
2. Declare and use an `algLast` to be more explicit about which algorithms must be first and last in the list (plus it rounds out the symmetry).

Reply:
Done.